bot/exts/info/resources.py

```diff
@@ -0,0 +1,74 @@
+import re
+from typing import Optional
+from urllib.parse import quote
+
+from discord import Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+
+REGEX_CONSECUTIVE_NON_LETTERS = r"[^A-Za-z0-9]+"
+RESOURCE_URL = "https://www.pythondiscord.com/resources/"
+
+
+def _to_kebabcase(resource_topic: str) -> str:
+    """
+    Convert any string to kebab-case.
+
+    For example, convert
+    "__Favorite FROOT¤#/$?is----LeMON???" to
+    "favorite-froot-is-lemon"
+    Code adopted from:
+    https://github.com/python-discord/site/blob/main/pydis_site/apps/resources/templatetags/to_kebabcase.py
+    """
+    # First, make it lowercase, and just remove any apostrophes.
+    # We remove the apostrophes because "wasnt" is better than "wasn-t"
+    resource_topic = resource_topic.casefold()
+    resource_topic = resource_topic.replace("'", '')
+
+    # Now, replace any non-letter that remains with a dash.
+    # If there are multiple consecutive non-letters, just replace them with a single dash.
+    # my-favorite-class is better than my-favorite------class
+    resource_topic = re.sub(
+        REGEX_CONSECUTIVE_NON_LETTERS,
+        "-",
+        resource_topic,
+    )
+
+    # Now we use strip to get rid of any leading or trailing dashes.
+    resource_topic = resource_topic.strip("-")
+    return resource_topic
+
+
+def to_kebabcase(resource_topic: str) -> str:
+    """Convert a string to kebab-case."""
+    return _to_kebabcase(resource_topic)
```

Comment:
This function can be removed, and `_to_kebabcase` renamed to `to_kebabcase`, there's not much point having a second function if all it does is call the first.

Reply:
I agree. I had a nice conversation with @mustafaquraish who explained when separating the logic out would make sense. It was very helpful! Just pushed the updated code for your review.